components/DeleteButton.tsx

```diff
@@ -0,0 +1,104 @@
+import { useRouter } from "next/router";
+import { Button, Text } from "@mantine/core";
+import { cleanNotifications, showNotification, NotificationProps } from "@mantine/notifications";
+import { Check, X } from "tabler-icons-react";
+import { useModals } from "@mantine/modals";
+
+/**
+ * DeleteButton is a component for rendering a delete resource button
+ * @returns Button that, when clicked, displays a notification verifying that the user
+ * wants to delete the resource. Only deletes if the user confirms, cancels otherwise.
+ */
+export default function DeleteButton() {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const router = useRouter();
+  const modals = useModals();
+  const { resourceType, id } = router.query;
+
+  let customMessage: NotificationProps["message"] = <div>Problem connecting to server</div>;
+  let notifProps: NotificationProps = {
+    message: customMessage,
+    color: "red",
+    icon: <X size={18} />,
+    autoClose: false,
+  };
+
+  const deleteHandler = () => {
```

Comment:
@natjoe4 suggested having async/await syntax over .then syntax in this function in order to prevent readability problems.

Reply:
I just talked to Nat and Matt, and they decided to leave it as is in for consistency with other docs

Reply:
Logical